kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -259,6 +301,198 @@ void shouldIncrementConnectionMetrics(KafkaCluster cluster, Topic topic) throws
             assertMetricsWithValue(updatedMetricsList, "kroxylicious_downstream_connections_total", null);
             assertMetricsWithValue(updatedMetricsList, "kroxylicious_upstream_connections_total", null);
         }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    void shouldIncrementUpstreamConnectionErrorMetrics(KafkaCluster cluster, Topic topic) {
+        var config = proxy(cluster)
+                .addToFilterDefinitions()
+                .addToDefaultFilters()
+                .withNewManagement()
+                .withNewEndpoints()
+                .withNewPrometheus()
+                .endPrometheus()
+                .endEndpoints()
+                .endManagement();
+
+        // Given
+        try (var tester = kroxyliciousTester(config);
+                var managementClient = tester.getManagementClient()) {
+            var metricsList = managementClient.scrapeMetrics();
+            assertMetricsDoesNotExist(metricsList, "kroxylicious_upstream_errors_total", null);
+
+            // When
+            var producer = tester.producer();
+
+            producer.send(new ProducerRecord<>(topic.name(), "key", "value")).get(100, TimeUnit.SECONDS);
```

Comment:
100ms is too short.  We'd probably see a sporadically failing test on some slow CI environments.

Also when dealing with Kafka futures in tests, our prefered approach is to use Assert-J.  That handles all the exceptions for us.

```
assertThat(producer.send(new ProducerRecord<>(topic.name(), "key", "value")))
.succeedsWith(Duration.ofSeconds(5));
```